karate-core/src/test/java/com/intuit/karate/core/feature-result.json

```diff
@@ -4,6 +4,7 @@
             "stepResults": [
                 {
                     "result": {
+                        "matchingMethod": "com.intuit.karate.ScenarioActions.print(java.lang.String) [\"'in background'\"]",
```

Comment:
We need the method to be able to compare on whether the execution should continue after a given step is executed and we decided to keep the reference in the StepResult instead of step (we only need java.lang.reflect.Method not StepRuntime.MethodMatch but used MethodMatch to keep things consistent with the framework).

To be able to re-hydrate we need to be able to export that entire class as JSON and reimport. At first I was trying just the simple toString() but there are too many edge cases as you can see with the unit tests that I did so ended up return the arguments section of that class using the Json.of(args) which already exports as a JSON string.

What we can do to make it more visually appealing is to nest that object instead of just shoving it as a string
"result" : {
  "matchingMethod" : {
    "methodSignature": "com.intuit.karate.ScenarioActions.print(java.lang.String)",
    "args": _the actual json generated thus avoiding the escaping characters_
}
}

Reply:
ok. I'm not worried about the visual part this is going to considerably bloat the JSON (that we plan to use in future for distributed test execution) considerably. is there a way we can avoid it. if it needs to go back into the `Step` consider me convinced now :)

Reply:
I kind of agree but now I already moved it haha and in terms of object model it does make sense to stay in StepResult.

What if I add some property that we can override somehow to exclude fields from the generated JSON (maybe  in the Suite or the Runner)? Make it excluded by default until the time we think we might need it.

Reply:
okay, I suck at reading code in PR. need to find a way to improve. but I will approve this (and perhaps tweak later), let me know

Reply:
Let me do that change first as I agree this is not currently adding value and it’s more anticipating a potential future use case.

Reply:
Not the prettiest solution... but I stuck a System Property in the `Result` that if evaluates to true it'll include that field in the output JSON. If we find the need for this to change in the future we change change how we configure that property.